mono/metadata/locales.c

```diff
@@ -515,7 +516,9 @@ get_darwin_locale (void)
 		CFRelease (locale);
 	}
 
-	return g_strdup (darwin_locale);
+	mono_memory_barrier ();
+	cached_locale = darwin_locale;
+	return g_strdup (cached_locale);
```

Comment:
As the assignment is not atomic, this should be `g_strdup (darwin_locale);`

Reply:
Pointer assignments are atomic.